### PR TITLE
Components: promote `Surface`

### PR DESCRIPTION
**Pull request by ciampo**
## Description

Promote the `Surface` component from the `packages/components/src/ui` to the `packages/components/src` folder.

- Update paths, imports and exports
- Refine Docs
- Add export to `@wordpress/components` as `__experimentalSurface`

## Testing instructions

- [ ] Project builds successfully
- [ ] All unit tests pass
- [ ] Build the project locally and make sure that the [new, local version](http://localhost:50240/?path=/story/components-experimental-surface--default) works as expected (for comparison, here is the [version that's currently on `trunk`](https://wordpress.github.io/gutenberg/?path=/story/g2-components-experimental-surface--default))

## Screenshots 

Screenshot 2021-06-03 at 20 01 37

## Types of changes

Refactor

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- N/A I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/components/src/surface/README.md [hunk 1]
```diff
@@ -24,48 +27,58 @@ function Example() {
...
+### `backgroundSize`: number
 
-**Type**: `number`
+- Required: No
+- Default: `12`
 
 Determines the grid size for "dotted" and "grid" variants.
 
-##### border
+### `border`: `boolean`
```
Comment (ciampo) [inline packages/components/src/surface/README.md]: I'm not personally sure about the APIs around `border`.

Currently, there are 5 different props:

- `border` shows the border on all four sides
- `border{Top,Left,Right,Bottom}` each show one individual border

It feels a bit weird to me that there is partial overlap between them — i.e. that if `border` is `true`, all other props become meaningless

File: packages/components/tsconfig.json [hunk 2]
```diff
@@ -44,7 +44,8 @@
 		"src/h-stack/**/*",
 		"src/v-stack/**/*",
 		"src/z-stack/**/*",
-		"src/view/**/*"
+		"src/view/**/*",
+		"src/surface/**/*"
```
Comment (tyxla) [inline packages/components/tsconfig.json:49]: Nit: seems like we don't quite follow an alphabetical sorting order here

File: packages/components/src/surface/README.md (hunk 1 above)
Comment (tyxla) [reply packages/components/src/surface/README.md]: Agreed, this doesn't make much sense to me, especially if we consider how borders behave in CSS. I guess it makes sense that the `border*` take precedence over `border`, but that might also require a change in the default values of those props.

Comment (tyxla) [reply packages/components/src/surface/README.md]: Regardless of what we decide here, it makes sense to document how all those border props behave, and which one takes precedence over the other(s).

**Review by tyxla — APPROVED** [review]
Looks good and tests well for me 👍

File: packages/components/src/surface/README.md (hunk 1 above)
Comment (ciampo) [reply packages/components/src/surface/README.md]: Probably the easiest solution that I see is to remove the `border` prop, and leave the remaining individual props as they are. That would avoid overlaps and only make setting a border around the whole surface a bit more verbose.

WDYT?

Comment (tyxla) [reply packages/components/src/surface/README.md]: I'd say I can see some value in having them separate, but I'd likely have the `border*` default to the current value of `border` of not specified. I think that would be easy, too, and would solve the problem, while still resembling how border specificity works in CSS. What do you think about that?

Comment (ciampo) [reply packages/components/src/surface/README.md]: As discussed with @tyxla , we decided to [remove the `border` prop for now](https://github.com/WordPress/gutenberg/pull/32439/commits/f9761ed4b8a95dfe2fc8ee72b956c80d47ad0237) and simplify the component's APIs

**Pull request merged**